src/geom/CoordinateSequence.cpp

```diff
@@ -288,7 +288,9 @@ CoordinateSequence::getOrdinate(std::size_t index, std::size_t ordinateIndex) co
         case CoordinateSequence::Z:
             return hasZ() ? getAt<Coordinate>(index).z : DoubleNotANumber;
         case CoordinateSequence::M:
-            return hasM() ? (hasZ() ? getAt<CoordinateXYZM>(index).m : getAt<CoordinateXYM>(index).m) : DoubleNotANumber;
+            return getCoordinateType() == CoordinateType::XYZM ? getAt<CoordinateXYZM>(index).m :
+                                                                 getCoordinateType() == CoordinateType::XYM ? getAt<CoordinateXYM>(index).m :
```

Comment:
Thanks for the input! I don't see any test coverage with `getCoordinateType() == CoordinateType::XYM`. The tests appear to only cover `CoordinateType::XYZM`.

Reply:
That's right. It's not possible to construct a `CoordinateSequence` that is backed by `XY` or `XYM` unless you change this `#define`:

https://github.com/libgeos/geos/blob/main/src/geom/CoordinateSequence.cpp#L40

The `main` version of line 291 is only correct if the define is _off_ (bad). The version you proposed is correct if the define is _on_, which it is (better). This version is correct either way.